src/adaptive.jl

```diff
@@ -0,0 +1,93 @@
+mutable struct Adaptor
+    accepted::Integer
+    total   ::Integer
+    tune    ::Integer   # tuning interval
+    target  ::Float64   # target acceptance rate
+    bound   ::Float64   # bound on logσ of Gaussian kernel
+    δmax    ::Float64   # maximum adaptation step
+end
+
+Adaptor(; tune=25, target=0.44, bound=10., δmax=0.2) =
+    Adaptor(0, 0, tune, target, bound, δmax)
+
+"""
+    AdaptiveProposal{P} 
+
+An adaptive Metropolis-Hastings proposal. In order for this to work, the
+proposal kernel should implement the `adapted(proposal, δ)` method, where `δ`
+is the increment/decrement applied to the scale of the proposal distribution
+during adaptation (e.g. for a Normal distribution the scale is `log(σ)`, so
+that after adaptation the proposal is `Normal(0, exp(log(σ) + δ))`).
+
+# Example
+```julia
+julia>  p = AdaptiveProposal(Uniform(-0.2, 0.2));
+
+julia> rand(p)
+0.07975590594518434
+```
+
+# References 
+
+Roberts, Gareth O., and Jeffrey S. Rosenthal. "Examples of adaptive MCMC."
+Journal of Computational and Graphical Statistics 18.2 (2009): 349-367.
+"""
+mutable struct AdaptiveProposal{P} <: Proposal{P}
+    proposal::P
+    adaptor ::Adaptor
+end
+
+function AdaptiveProposal(p; kwargs...) 
+    AdaptiveProposal(p, Adaptor(; kwargs...))
```

Comment:
```suggestion
    return AdaptiveProposal(p, Adaptor(; kwargs...))
```

We typically specify the `return` command just as a style preference.